virttest/ip_sniffing.py

```diff
@@ -0,0 +1,311 @@
+                            remote host, otherwise on local host.
+        """
+        self._cache = addr_cache
+        self._logfile = log_file
+        self._remote_opts = remote_opts
+        self._process = None
+
+    @classmethod
+    def _is_supported_remote(cls, session):
+        return session.cmd_status("which %s" % cls.command) == 0
+
+    @classmethod
+    def is_supported(cls, session=None):
+        """
+        Check if host supports the sniffer.
+
+        :param session: Remote host session. If provided, performs the
+                        check on remote host, otherwise on local host.
+        """
+        if session:
+            return cls._is_supported_remote(session)
+        try:
+            utils_path.find_command(cls.command)
+            return True
+        except utils_path.CmdNotFoundError:
+            return False
+
+    def _output_handler(self, line):
+        raise NotImplementedError
+
+    def _output_logger_handler(self, line):
+        try:
+            log_line(self._logfile, line)
+        except Exception as e:
+            logging.warn("Can't log ip sniffer output: '%s'", e)
+        self._output_handler(line)
+
+    def _start_remote(self):
+        address, port, username, password, prompt = self._remote_opts
+        cmd = "%s %s" % (self.command, self.options)
+        logging.debug("Run '%s' on host '%s'", cmd, address)
+        login_cmd = ("ssh -o UserKnownHostsFile=/dev/null "
+                     "-o StrictHostKeyChecking=no "
+                     "-o PreferredAuthentications=password -p %s %s@%s" %
+                     (port, username, address))
+
+        self._process = aexpect.ShellSession(
+            login_cmd,
+            output_func=self._output_logger_handler)
+        handle_prompts(self._process, username, password, prompt)
+        self._process.sendline(cmd)
+
+    def _start(self):
+        cmd = "%s %s" % (self.command, self.options)
+        if self._remote_opts:
+            self._start_remote()
+        else:
+            self._process = aexpect.run_tail(
+                command=cmd,
+                output_func=self._output_logger_handler)
+
+        if not self._process.is_alive():
+            logging.warn("Could not start ip sniffer (%s)", self.command)
+            logging.warn("Status: %s", self._process.get_status())
+            msg = format_str_for_message(self._process.get_output())
+            logging.warn("Output: %s", msg)
+
+    def is_alive(self):
+        """Check if the sniffer is alive."""
+        if not self._process:
+            return False
+        return self._process.is_alive()
+
+    def start(self):
+        """Start sniffing."""
+        if self.is_alive():
+            return
+        self.stop()
+        self._start()
+
+    def stop(self):
+        """Stop sniffing."""
+        if self._process:
+            self._process.close()
+            old_p = self._process
+            self._process = None
+            del old_p
```

Comment:
these three lines should be same with just `self._process = None`

Reply:
OK, make sense, let's leave this gc stuff to python.